File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/auth/dtos/RegisterCompanyRequest.java
package com.slotify.backend.spring.auth.dtos;

import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class RegisterCompanyRequest {
    @NotBlank
    @Size(max = 200)
    private String companyName;

    @NotBlank
    @Size(min = 10, max = 16)
    private String password;

    @Email
    @Size(max = 320)
    private String emailAddress;

    @Min(1)
    private Integer defaultMaxConcurrentServices;

    @NotBlank
    @Size(max = 300)
    private String physicalAddress;

}

File: app/src/main/java/com/raj/slotify/dtos/company/RegisterCompanyRequest.kt
package com.example.slotify.dtos.company

data class RegisterCompanyRequest(
    var companyName: String? = null,
    var password: String? = null,
    var emailAddress: String? = null,
    var defaultMaxConcurrentServices: Int? = null,
    var physicalAddress: String? = null
)